services/webnn/ort/graph_builder_ort.cc

```diff
@@ -745,6 +748,98 @@ GraphBuilderOrt::AddInstanceNormalizationOperation(
   return base::ok();
 }
 
+[[nodiscard]] base::expected<void, mojom::ErrorPtr>
+GraphBuilderOrt::AddLayerNormalizationOperation(
+    const mojom::LayerNormalization& layer_normalization) {
+  const OperandDataType input_data_type =
+      GetOperand(layer_normalization.input_operand_id).descriptor.data_type();
+
+  const std::string input_name =
+      GetOperandName(layer_normalization.input_operand_id);
+  std::vector<const char*> input_names = {input_name.c_str()};
+
+  auto axes = layer_normalization.axes;
+  std::sort(axes.begin(), axes.end());
+  const auto axes_size = axes.size();
+  const std::vector<uint32_t>& input_shape =
+      GetOperand(layer_normalization.input_operand_id).descriptor.shape();
+  if (axes[axes_size - 1] != input_shape.size() - 1 ||
+      axes[0] != input_shape.size() - axes_size) {
+    return NewNotSupportedError(
+        "Onnx LayerNormalization only supports last consecutive dimensions as "
+        "axes.");
+  }
+  uint32_t axis = axes[0];
+
+  std::string scale_name;
+  base::CheckedNumeric<uint32_t> checked_scale_size =
+      std::accumulate(input_shape.begin() + axis, input_shape.end(),
+                      base::CheckedNumeric<uint32_t>(1), std::multiplies());
+  if (!checked_scale_size.IsValid()) {
+    return NewNotSupportedError("The size of scale is too large.");
+  }
+  std::vector<float> scale_data(checked_scale_size.ValueOrDie(), 1.0f);
```

Comment:
Can scale be scalar? ONNX doc mentions it supports unidirectional broadcasting?

https://onnx.ai/onnx/operators/onnx__LayerNormalization.html

Reply:
I suppose onnx layer norm can't support empty axes since the axis must be set to [-r, r).  In this PR, I throw an error when axes is empty. Maybe we can use reduceMean to emulate in the future. WDYT? @huningxin

Reply:
> Can scale be scalar? 

In what case would scale be a scalar? Could you please give an example.

Reply:
> > Can scale be scalar?
> 
> In the blink side, we check the shape of the scale must be the same as the corresponding input shape.

Reply:
> > > Can scale be scalar?
> > 
> > 
> > In the blink side, we check the shape of the scale must be the same as the corresponding input shape.

I misunderstand your comment. Please ignore my reply. I will try if scalar scale can work.

Reply:
> Can scale be scalar? ONNX doc mentions it supports unidirectional broadcasting?
> 
> https://onnx.ai/onnx/operators/onnx__LayerNormalization.html

It seems we can't set scalar here.  If I set scalar scale, ort will report this error:

`LayerNormalization node. Name:'_0' Status Message: Size of X.shape()[axis:] == 12. Size of scale and bias (if provided) must match this. Got scale size of 1 and bias size of 12
 base::debug::CollectStackTrace [0x00007FFEB79F8EB5+21] (C:\workspace\chromium\src\base\debug\stack_trace_win.cc:327)`.

Reply:
> I suppose onnx layer norm can't support empty axes since the axis must be set to [-r, r). In this PR, I throw an error when axes is empty. Maybe we can use reduceMean to emulate in the future. WDYT? @huningxin

We may handle empty axes by `output = bias + (scale * 0)`, similar to CoreML backend does https://source.chromium.org/chromium/chromium/src/+/main:services/webnn/coreml/graph_builder_coreml.cc;l=3167?q=services%2Fwebnn%2Fcoreml%2Fgraph_builder_coreml.cc

Reply:
> > I suppose onnx layer norm can't support empty axes since the axis must be set to [-r, r). In this PR, I throw an error when axes is empty. Maybe we can use reduceMean to emulate in the future. WDYT? @huningxin
> 
> We may handle empty axes by `output = bias + (scale * 0)`, similar to CoreML backend does https://source.chromium.org/chromium/chromium/src/+/main:services/webnn/coreml/graph_builder_coreml.cc;l=3167?q=services%2Fwebnn%2Fcoreml%2Fgraph_builder_coreml.cc

Handle empty axes by `output = bias + (scale * 0)`. Please take another look. Thanks!